src/plugins/intel_npu/src/plugin/npuw/partitioning/patterns/opt.cpp

```diff
@@ -600,9 +607,12 @@ DQMatMulGQ2iP::DQMatMulGQ2iP(Context::Ref ctx) {
             matched_qcoeff->set_partial_shape(ts_shape);
             matched_qcoeff->validate_and_infer_types();
 
+            // Select proper activation shape
+            std::size_t act_dim_shape = act_shape[0] > act_shape[1] ? act_shape[0] : act_shape[1];
```

Comment:
this code is just `std::max()`, but I'd expect you to pick a dim(ension), not the dimension value - and use it as `act_shape[d]` instead of `act_shape[1]` that was before

Reply:
Done